src/orchestrator.rs

```diff
@@ -162,50 +178,98 @@ fn get_chunker_ids(
+
+    // Create generation client
+    if let Some(generation) = &config.generation {
+        match generation.provider {
             GenerationProvider::Tgis => {
-                let client = TgisClient::new(
-                    clients::DEFAULT_TGIS_PORT,
-                    &[(COMMON_ROUTER_KEY.to_string(), generation.service.clone())],
+                let client = create_grpc_client(
+                    DEFAULT_TGIS_PORT,
+                    &generation.service,
+                    GenerationServiceClient::new,
                 )
                 .await;
-                GenerationClient::tgis(client)
+                let tgis_client = TgisClient::new(client);
+                let generation_client = GenerationClient::tgis(tgis_client);
+                clients.insert("generation".to_string(), generation_client);
             }
             GenerationProvider::Nlp => {
-                let client = NlpClient::new(
-                    clients::DEFAULT_CAIKIT_NLP_PORT,
-                    &[(COMMON_ROUTER_KEY.to_string(), generation.service.clone())],
+                let client = create_grpc_client(
+                    DEFAULT_NLP_PORT,
+                    &generation.service,
+                    NlpServiceClient::new,
                 )
                 .await;
-                GenerationClient::nlp(client)
+                let health_client =
+                    create_grpc_client(DEFAULT_NLP_PORT, &generation.service, HealthClient::new)
+                        .await;
+                let nlp_client = NlpClient::new(client, health_client);
+                let generation_client = GenerationClient::nlp(nlp_client);
+                clients.insert("generation".to_string(), generation_client);
             }
-        },
-        None => GenerationClient::not_configured(),
-    };
-    // TODO: simplify all of this
-    let chunker_config = match &config.chunkers {
-        Some(chunkers) => chunkers
-            .iter()
-            .map(|(chunker_id, config)| (chunker_id.clone(), config.service.clone()))
-            .collect::<Vec<_>>(),
-        None => vec![],
-    };
-    let chunker_client = ChunkerClient::new(clients::DEFAULT_CHUNKER_PORT, &chunker_config).await;
-
-    let detector_config = config
-        .detectors
-        .iter()
-        .map(|(detector_id, config)| (detector_id.clone(), config.service.clone()))
-        .collect::<Vec<_>>();
-    let detector_client =
-        DetectorClient::new(clients::DEFAULT_DETECTOR_PORT, &detector_config).await;
-
-    (generation_client, chunker_client, detector_client)
+            GenerationProvider::OpenAi => unimplemented!(),
+        }
+    }
+
+    // Create chat generation client
+    if let Some(chat_generation) = &config.chat_generation {
+        match chat_generation.provider {
+            GenerationProvider::OpenAi => {
+                let client =
+                    create_http_client(DEFAULT_OPENAI_PORT, &chat_generation.service).await;
```

Comment:
Q: Is it possible to push the detail that this client needs `http_client` to its own module? 

so that we can just do `client = GenerationProvider:OpenAI::get_client(DEFAULT_OPENAI_PORT, &chat_generation.service)` 

that way the creation of client from this file for all, looks same?

Reply:
This change to inner client dependency injection, rather than creating the client in `<client_type>::new()`, was intentional for testing purposes.

I'm not sure I fully understand your suggestion though, so let's discuss during standup.

Reply:
What I meant is, each of the client module, like `src/clients/detector/text_chat.rs` or `nlp.rs`, knows if they can work with grpc or http.

So am thinking, the previous paradigm that created client inside these module was nice from `orchestratrator.rs` perspective, since it just needed to initialize the nlp client and it would call the `create_grpc_client` function etc. 

With this change, this information that a client is gRPC or http, now lives in 2 places.

So my question was, if we can move invocation of `create_grpc_client` (or `create_clients`)  inside the same module as the client and expose just a more consistent `get_client` function over here.

So we need not call this `crate_grpc_client` or `create_clients` function from inside the constructor like before, but just add that as a static method or something.


So what if it was something like:
```rust
impl NlpClient {
    pub fn new(...) {...}  
    pub fn create_client(...) -> Self {
     // call to create_grpc_client
   }
```

and this would potentially solve the testing problem you were seeing and at the same time keep the client type (grpc vs http) information inside the module 


thoughts?

Reply:
```
TgisClient::new(client);
```
The way I see it, a client and its inner client are highly coupled, so i think this is a good point to bring up, since the creation process requires making the inner client in a separate step.

On friday I was trying to make the inner client part of the client trait itself:
```rust
#[async_trait]
pub trait Client: Send + Sync + 'static {
    type Inner;

    /// Returns the inner client.
    fn client(&self) -> Self::Inner;
    ...
```
This way it could all conform to a single trait. This runs into type erasure complexity because we are aggregating a hashmap of client impls with different `Inner` client types, we would need an `AnyClient` to use in place of `Client<Inner = ...>`. Even though going down this route is probably too much for now, I still think the client and its inner client sound highly coupled so we could hide the inner service as @gkumbhat suggests.

Reply:
Just to clarify, the rationale for moving inner client construction out of `<client>::new()` is to easily enable injecting mock http/grpc clients which are not created through `create_http_client()` and `create_grpc_client()`. Decoupling this provides greater flexibility for testing purposes.

Reply:
> injecting mock http/grpc clients which are not created through create_http_client() and create_grpc_client()

Thinking about this further, I don't think this will be necessary. I'll revert this back and we can adjust if needed once we figure out our mocking strategy.